examples/tv-casting-app/android/App/app/src/main/java/com/matter/casting/DiscoveryExampleFragment.java

```diff
@@ -0,0 +1,400 @@
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.ListView;
+import android.widget.TextView;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import com.R;
+import com.matter.casting.core.CastingPlayer;
+import com.matter.casting.core.CastingPlayerDiscovery;
+import com.matter.casting.core.MatterCastingPlayerDiscovery;
+import com.matter.casting.support.MatterError;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+
+public class DiscoveryExampleFragment extends Fragment {
+  private static final String TAG = DiscoveryExampleFragment.class.getSimpleName();
+  // 35 represents device type of Matter Casting Player
+  private static final int DISCOVERY_TARGET_DEVICE_TYPE = 35;
+  private static final int DISCOVERY_RUNTIME_SEC = 15;
+  private TextView matterDiscoveryMessageTextView;
+  private static final ScheduledExecutorService executorService =
+      Executors.newSingleThreadScheduledExecutor();
+  private ScheduledFuture scheduledFutureTask;
+  private static final List<CastingPlayer> castingPlayerList = new ArrayList<>();
+  private static ArrayAdapter<CastingPlayer> arrayAdapter;
+
+  // Get a singleton instance of the MatterCastingPlayerDiscovery
+  private static final CastingPlayerDiscovery matterCastingPlayerDiscovery =
+      MatterCastingPlayerDiscovery.getInstance();
+
+  /**
+   * Implementation of a CastingPlayerChangeListener used to listen to changes in the discovered
+   * CastingPlayers
+   */
+  private static final CastingPlayerDiscovery.CastingPlayerChangeListener
+      castingPlayerChangeListener =
+          new CastingPlayerDiscovery.CastingPlayerChangeListener() {
+            private final String TAG =
+                CastingPlayerDiscovery.CastingPlayerChangeListener.class.getSimpleName();
+
+            @Override
+            public void onAdded(CastingPlayer castingPlayer) {
+              if (castingPlayer != null) {
+                Log.i(
+                    TAG,
+                    "onAdded() Discovered CastingPlayer deviceId: " + castingPlayer.getDeviceId());
+                // Display CastingPlayer info on the screen
+                new Handler(Looper.getMainLooper())
+                    .post(
+                        () -> {
+                          final Optional<CastingPlayer> playerInList =
+                              castingPlayerList
+                                  .stream()
+                                  .filter(node -> castingPlayer.equals(node))
+                                  .findFirst();
+                          if (playerInList.isPresent()) {
+                            Log.d(
+                                TAG,
+                                "onAdded() Replacing existing CastingPlayer entry "
```

Comment:
Why would we need to replace an existing CastingPlayer when onAdded() is called?

Reply:
Removed. Not needed since duplicate players are filtered out upstream.